lib/rucio/gateway/subscription.py

```diff
@@ -131,6 +131,9 @@ def update_subscription(
                 else:
                     for rule in metadata['replication_rules']:
                         validate_schema(name='activity', obj=rule.get('activity', 'default'), vo=vo)
+            if 'state' in metadata and metadata['state'] is not None:
+                metadata['state'] = SubscriptionState(metadata['state'])
```

Comment:
Out of scope (for now we always pass a valid state from client): this should validate that the state passed in the metadata is a valid SubscriptionState. We could just add a try/except for `ValueError` here, e.g. if an invalid value is passed then:

```
ValueError: 'C' is not a valid SubscriptionState
```

I created this issue to track this: https://github.com/rucio/rucio/issues/7820